### PR TITLE
Make SPA threadsafe again

### PR DESCRIPTION
**Pull request by dguittet**
fix https://github.com/NREL/SAM/issues/1882 by moving the SPA table inside irradproc

### CODE REVIEW

**Review by cpaulgilman — APPROVED** [review]
Thank you for MSTA!

**Comment by dguittet** [comment]
> Just for future reference, do you have any results showing the final impact on speed?

I fixed the speed issue-- it was due to not initializing the ascension and declination to {0, 0}, so the final speed impacts are the same!

File: shared/lib_irradproc.h [hunk 1]
```diff
@@ -602,6 +602,48 @@ double rts_sun_altitude(double latitude, double delta_prime, double h_prime);
...
+    spa_table_key(double j, double dt, double p, double t, double a, double d);
+};
+
+template <>
+struct std::hash<spa_table_key>
+{
+  std::size_t operator()(const spa_table_key& k) const;
+};
+
+class solarpos_lookup
```
Comment (sjanzou) [inline shared/lib_irradproc.h:631]: Thanks for updating...

**Review by sjanzou — APPROVED** [review]
Working for the parametrics now! Thanks for updating

**Pull request merged**